### DISCUSSION TITLE
Welcome to pcrm-book Discussions!

### DISCUSSION THREAD

**Discussion by antonvorobets** [fortitudo-tech/pcrm-book#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of this community. 

  I hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded.

  To get started, comment below with an introduction of yourself and tell us about what you want to achieve with this book.

**Comment by Hitz1690** [comment]
Hi Anton,

I have 2 broad approaches to Investing in the equity markets, 

1) FUNDAMENTAL / TECHNICAL INVESTING IN EQUITIES WHICH IS DELIVERY BASED (Time frame: - Few days to 2 yr / 3 yr / 5 yrs etc)

2) DERIVATIVES BASED IN INDEXES AND SCRIPTS (Intraday)

Please find some questions below. 

1) General Application of the Framework for Real-World Usage
Where do you see the framework being most impactful in real-world applications? Are there specific areas in finance or trading where it has proven to add significant value?

2) Flexibility Across Markets
How adaptable is the framework to different asset classes such as equities, commodities, cryptocurrencies, or fixed income?

3) Short-Term vs. Long-Term
Is the framework better suited for short-term trading strategies or long-term portfolio management? Can it be effectively adapted for both?

4) Beyond Trading
Can the framework be applied outside of direct trading, such as in portfolio risk management, algorithmic strategy design, or even corporate financial planning?

5) Practical Challenges
What are the most common challenges practitioners face when trying to implement this framework in real-world trading?

6) Integration with Technology
How does the framework complement modern trading technologies, such as algorithmic trading systems or machine learning models?

7) Customizing the Framework
To what extent can practitioners modify the framework to suit their own strategies or market conditions without compromising its core principles?

8) Backtesting and Validation
What are the best practices for backtesting and validating strategies built using this framework?

9) Scalability
Does the framework scale well for high-frequency trading, or is it more effective for lower-frequency, larger trades?
Implementation

For someone just starting to implement the framework, what initial steps or simplified applications would you recommend?

Best Regards,
Hitesh.

**Reply by antonvorobets** [reply]
Hi Hitesh,

Thanks for these great questions and apologies for the late reply. I have been extremely busy the last couple of days.

The framework is feasible for both of the approaches you mention. The portfolio management framework for derivatives with a separation between market value and exposure makes derivatives handling as easy as plain vanilla cash instruments. You of course need to do some handling of margins for derivatives trades.

Here are answers to the specific questions:

1. The framework can be used for all markets and should be feasible for all horizons, perhaps except for the most high-frequent. My own background is within asset allocation including derivatives and tail risk hedging strategies. These kind of investors find the framework very natural compared to how they work and the problems they have.

2. There is no asset class limitation. Once we transform the raw data into something that is stationary, our statistical market simulation models do not care what we call the market :-)

3. It is useful for both with short-term being something like one day. Intraday above the most high-frequent (let's say from 10 minute holding periods and up) should also be feasible, but I have not tried it in these situations.

4. Yes, portfolio risk follows almost immediately from the market simulation. Many of the methods are designed to combine the data with discretionary inputs, but these methods can also be applied in a systematic way, while I don't have personal experience using them in this way. I am not familiar enough with corporate financial planning to answer if it can be used there or not.

5. Fast and stable implementations. One of the main reasons that fully general CVaR has not been adopted more widely is the belief that these problems are too hard to solve in practice. However, algorithms that are fast and stable do exist, but these are hard to discover and properly implement.

6. I am not aware of any integration issues, so it should work just fine.

7. In the book, I present the framework as the general princinples for a good investment calculator, which I think is an accurate representation. People can use the methods however they see fit. The examples are just some suggestions of common use to solve common problems. Note that there is a great deal of flexibility in terms of how views are specified, Bayesian networks defined, etc.

8. I think the same principles as for validating and backtesting all other strategies and methods. I would advice to always generate a synthetic market simulation and validate strategies based on this. Relying just on historical realizations is dangerous. See this example: https://open.substack.com/pub/antonvorobets/p/tactical-asset-allocation-performance

9. If the trading is very high-frequent (let's say below 10 minutes), I am afraid that even the fastest implementations will be too slow. Otherwise, it should be feasible with a good implementation.

I would generally not recommend that people try implementing production quality implementations unless they absolutely have to. It is very hard. But a good starting point is the fortitudo.tech Python package. It contains perhaps 2%-3% of the easier work.

**Comment by Hitz1690** [comment]
Hi Anton,

Thank you for providing such detailed and insightful answers to the questions. Your framework for stylized market facts is truly fascinating, and I’ve been exploring how best to apply it in my trading strategies.

I understand that the framework likely originated from addressing specific pain points or challenges in trading. I would love to hear your perspective on this:

What were the key pain points or scenarios you had in mind while developing this framework?

In your view, in which type of trading would this framework have the most impact? Would it be in delivery-based trading, where stocks are bought and sold for profit over varying timelines, or in derivatives trading involving indexes or individual securities?

Additionally, given your background in derivatives and tail risk hedging strategies, could you provide an example of a specific strategy where you see this framework delivering substantial value? For instance, could it be used effectively in a delta-neutral options strategy, a tail-risk hedging mechanism, or perhaps in structuring trades to capture volatility arbitrage?

Your insights would be immensely helpful as I aim to align my approach more closely with the principles and strategies you have articulated.

Thank you once again for your time and feedback.

Warm regards,
Hitesh

**Reply by antonvorobets** [reply]
Hi Hitesh,

Thank you for the kind feedback.

The framework tries to address the many fundamental pain points that mainstream academic approaches don't.

The first one being **the separation between market values and exposures**. The idea came from when I worked with equity derivatives for a large Nordic bank. All the academic material was talking about portfolio weights as a measure of exposure, but this is quite meaningless for derivatives, for example, think about a futures position that has 0 market value but an exposure depending on the direction of the trade.

Because people are so used to market weights, they would introduce auxiliary offsetting cash positions to make everything sum to 1, but there are many issues with this approach in practice, so it is much easier to treat derivatives as they are with a separation between market value and exposure, see Section 6.1.

The second big paint point is **the academic reliance on the covariance matrix and mean-variance**. This is especially meaningless for derivatives where the nonlinearities are present by construction. However, I also found that asset allocation investors would often talk about how dependencies are different in normal and tail risk scenarios, which the empirical analysis also supports. So, it does not make sense to use mean-variance assuming that dependencies are linear and constant.

If you analyze fully general market scenarios, the dependencies can be arbitrarily complex. So having this starting point solves many issues, but it is of course harder to generate and analyze.

Since we focus on generating new paths for the relevant risk factors, such as, implied volatilities, interest rates, and the underlying, we can analyze whichever strategy we want, including dynamic delta-hedged ones (I will add a section on this, thanks for the inspiration).

Another big pain point is **consistent views/stress-testing**, which also used to rely on the mean-variance framework with the Black-Litterman model. This model has many issues, including all the above, but it also does not allow us to implement views and stress-tests in a consistent way for our derivatives P&L. This is one of the elegant features of Entropy Pooling.

I will write more about **tail risk hedging** in Section 7.3, but the main point is that outright hedging of a strategic basis is often too expensive (as the volatility risk premium from Section 2.3 illustrates), so the idea is to identify the scenarios where key diversification assumptions fails and build tail risk hedges that work in these. This is of course not a trivial task, but Entropy Pooling can help us, see for example Section 7.2.

In relation to **trading signals and strategies**, this is something that investors must identify themselves for the markets that they trade in. The book's purpose is rather to give a good quant / quantamental framework for analyzing and managing these. I personally think that one should be cautious when people share alpha signals and strategies. These are usually quite elusive and hard to come by, so if I do, I will probably trade them myself, while using the investment framework from the book :-)

I hope the above gave you some meaningful answers. But feel free to ask again.

Best,
Anton
